zkwrapper/source/zkwrapper.cc

```diff
@@ -90,20 +94,28 @@ ZKWrapper::ZKWrapper(std::string host, int &error_code, std::string root_path) {
 	}
 	init = 1;
 	if (root_path.size() != 0) {
-		bool root_exists;
-		if (!exists(root_path, root_exists, error_code)){
-			LOG(ERROR) << CLASS_NAME <<  "Failed to check if root directory " << root << " exists " << error_code;
-			init = -1;
-			return;
+
+		bool root_exists = false;
+		while (!root_exists) {
```

Comment:
This is sketchy but basically with multiple ZNodes or real environments, often times the initialization can fail for a number of times, before we get connected to ZK. Thus this is basically a blocking method waiting on a valid connection to ZK before allowing the ZK_client to be constructed.